### ISSUE TITLE
call component when data is ready

### ISSUE THREAD

**Issue by maximhoffman** [reactjs/react-rails#151]
I'm trying to call component in render function, when the data is ready, or return null. I don't know why, but react-rails gem return Parse Error in this string. Source code is below:

```javascript
var PlayersData = React.createClass({

  loadDataFromServer: function() {
    console.log("loadDataFromServer");
    $.ajax({
      url: this.props.url,
      dataType: 'json',
      success: function(data) {
        this.setState({data: data});
      }.bind(this),
      error: function(xhr, status, err) {
        console.error(this.props.url, status, err.toString());
      }.bind(this)
    });
  },

  getInitialState: function() {
    console.log("getInitialState");
    return {data: {}};
  },

  componentWillMount: function() {
    console.log("componentWillMount");
    this.loadDataFromServer();
  },

  render: function() {
    console.log("render");
    return (
      {this.state.data ?  <PlayersRepresentation data={this.state.data} /> : null}
    );
  }
});
```

Error trace log:

```
Showing /Users/user/Documents/Projects/rails_react/app/views/layouts/application.html.erb where line #12 raised:

Error: Parse Error: Line 30: Unexpected token .
  (in /Users/user/Documents/Projects/rails_react/app/assets/javascripts/components/match_center/players_data.js.jsx)
```

How does fix it?

**Comment by xionon** [comment]
Hi! Glad you're using react-rails. For future reference, this question is more appropriate for [stack overflow](http://stackoverflow.com/questions/tagged/reactjs), since it's a bug in your code, not react-rails itself.

If you look at the error you pasted, it tells you where the problem is:

> Error: Parse Error: Line 30: Unexpected token .

Line 30 is this line: `{this.state.data ?  <PlayersRepresentation data={this.state.data} /> : null}`

There are a couple problems here. You're getting a syntax error because you can't nest `{}` segments like this, it doesn't make any sense. Looking at your program logic, there's no case where `this.state.data` would ever return `false` or `null`, since `getInitialState` returns it as an empty object; therefore, the ternary operator is completely unnecessary. Lastly, it's not "react-y" to do this; you should always return an element, and you should always let that element decide what or how much to render based on passed-in props.

tl;dr your return statement should look like this:

`return (<PlayersRepresentation data={this.state.data} />)`

**Comment by zpao** [comment]
I think it would be ok to conditionally return null (though in this case I agree that letting PlayersRepresentation make that call is probably better).

Anyway, there's no bug with React nor react-rails here, just a JS syntax error.

**Comment by maximhoffman** [comment]
@xionon @zpao Maybe you are right, but i wrote my project by official tutorial. Anyway, i change getInitialState function on data: null, but rails return Parse Error again:
`Error: Parse Error: Line 30: Unexpected token . `
When i remove braces, everything is ok.
This is reason why i wrote in react-rails gem issue.
